src/components/panningViews/panToDismissView.js

```diff
@@ -0,0 +1,301 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Animated} from 'react-native';
+import {BaseComponent} from '../../commons';
+import {Constants} from '../../helpers';
+import asPanViewConsumer from './asPanViewConsumer';
+import PanningProvider from './panningProvider';
+
+const DEFAULT_SPEED = 20;
+const DEFAULT_BOUNCINESS = 6;
+const DEFAULT_DISMISS_ANIMATION_DURATION = 280;
+
+/**
+ * @description: PanToDismissView component created to making listening to swipe and drag events easier,
+ * Has to be used as a child of a PanningProvider that also has a PanListenerView
+ */
+class PanToDismissView extends BaseComponent {
+  static displayName = 'PanToDismissView';
+  static propTypes = {
+    /**
+     * The directions of the allowed pan (default allows all directions)
+     * Types: UP, DOWN, LEFT and RIGHT (using PanningProvider.Directions.###)
+     */
+    directions: PropTypes.arrayOf(PropTypes.oneOf(Object.values(PanningProvider.Directions))),
+    /**
+     * onDismiss callback
+     */
+    onDismiss: PropTypes.func,
+    /**
+     * The animation speed (default is 20)
+     */
+    animationSpeed: PropTypes.number,
+    /**
+     * The animation bounciness (default is 6)
+     */
+    animationBounciness: PropTypes.number,
+    /**
+     * The dismiss animation duration (default is 280)
+     */
+    dismissAnimationDuration: PropTypes.number,
+  };
+
+  static defaultProps = {
+    directions: [
+      PanningProvider.Directions.UP,
+      PanningProvider.Directions.DOWN,
+      PanningProvider.Directions.LEFT,
+      PanningProvider.Directions.RIGHT,
+    ],
+    animationSpeed: DEFAULT_SPEED,
+    animationBounciness: DEFAULT_BOUNCINESS,
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      deltaX: 0,
```

Comment:
maybe change to a clearer name like `animTranslateX` 
"anim" so it easy to tell it's an animated value, 
"translate" cause that's what it controls, also, we have here so many vars named "delta" it's getting a little confusing.

Reply:
I did want to change that but forgot, done.